### PR TITLE
chore: update staking helper version

### PR DESCRIPTION
**Pull request by defispartan**
## General Changes

- Updates implementation of UiStakeDataProvider helper contract to account for Safety Module 1.5 updates, and general cleanup to the helper contract
- The staking cooldown now applies to a specific amount, if the wallet balance > cooldown amount there is now the option to re-activate cooldown for the full wallet balance

## Testing

Paste into browser console -> reload page to enter fork mode

```
  localStorage.setItem('forkEnabled', 'true');
  localStorage.setItem('forkBaseChainId', '1');
  localStorage.setItem('forkNetworkId', '1');
  localStorage.setItem('forkRPCUrl', 'https://rpc.tenderly.co/fork/a654d0aa-04ef-4ff7-b827-8acb51c8f163');
```

Can simulate this address:

`0x342d8a2B1204894B18548320e73c84E96880c339`

which has a cooldown active and balance > cooldown amount. Or to test yourself can add the Tenderly RPC to a wallet, use this [tool](https://get-tenderly-balance.pages.dev/) to get ETH -> swap to stkAave on app.uniswap.org 

## Developer Notes

- Removes hardcoded staking types, and imports them from utilities instead
- Utilities version: https://github.com/aave/aave-utilities/pull/542
- UiStakeDataProvider version: https://github.com/aave/aave-stake-v2/pull/30
- User and safety module total balance use redeemable amount in place of balanceOf to automatically account for slashing if an event occurs

---

## Author Checklist

Please ensure you, the author, have gone through this checklist to ensure there is an efficient workflow for the reviewers.

- [x]  The base branch is set to `main`
- [x]  The title is using [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) formatting
- [ ]  The Github issue has been linked to the PR in the Development section
- [x]  The General Changes section has been filled out
- [x]  Developer Notes have been added (optional)

**If the PR is ready for review:**

- [x]  The PR is in `Open` state and not in `Draft` mode
- [x]  The `Ready for Dev Review` label has been added

## Reviewer Checklist

Please ensure you, as the reviewer(s), have gone through this checklist to ensure that the code changes are ready to ship safely and to help mitigate any downstream issues that may occur.

- [ ]  End-to-end tests are passing without any errors
- [ ]  Code style generally follows existing patterns
- [ ]  Code changes do not significantly increase the application bundle size
- [ ]  If there are new 3rd-party packages, they do not introduce potential security threats
- [ ]  If there are new environment variables being added, they have been added to the `.env.example` file as well as the pertinant `.github/actions/*` files
- [ ]  There are no CI changes, or they have been approved by the DevOps and Engineering team(s)
- [ ]  Code changes have been quality checked in the ephemeral URL
- [ ]  QA verification has been completed
- [ ]  There are two or more approvals from the core team
- [ ]  Squash and merge has been checked

### CODE REVIEW

**Comment by MareskoY** [comment]
@defispartan, everything is good except for one small bug with the ABPT icon:

Screenshot 2023-05-23 at 12 46 38

Additionally, I have added test coverage for the recall cooldown.

**Comment by MareskoY** [comment]
QC Passed

File: src/components/transactions/Stake/StakeModalContent.tsx [hunk 1]
```diff
@@ -61,8 +61,8 @@ export const StakeModalContent = ({ stakeAssetName, icon }: StakeProps) => {
   // staking token usd value
   const amountInUsd =
     Number(amount) *
-    (Number(normalize(stakeData?.stakeTokenPriceEth || 1, 18)) /
-      Number(normalize(stakeGeneralResult?.usdPriceEth || 1, 18)));
+    (Number(normalize(stakeData?.stakeTokenPriceEth || 1, 18)) *
```
Comment (grothem) [inline src/components/transactions/Stake/StakeModalContent.tsx:64]: I think this should stay as divide by

Comment (defispartan) [reply src/components/transactions/Stake/StakeModalContent.tsx:64]: ClaimUSD
UnstakeUSD

Fixed in both places

**Review by grothem — APPROVED** [review]

**Review by JoaquinBattilana — APPROVED** [review]

**Pull request merged**